src/screens/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,198 @@
+import { useState } from "react";
+import { View, Text, Image } from "react-native";
+import uuid from "react-native-uuid";
+import { useForm } from "@hooks/useForm";
+import AsyncStorageManager from "@storage/AsyncStorageManager";
+import TextField from "@components/TextField";
+import RadioButton from "@components/RadioButton";
+import GestureDetector from "@components/GestureDetector";
+import TextButton from "@components/TextButton";
+import { initialSignUpData as initialSignUpData, signUpValidations } from "@constants/constants";
+import styles from "./styles";
+import { SignInBackground as SignUpBackground } from "@src/assets/images";
+
+const SignUp = ({ navigation: { replace } }: SignUpScreenParamList) => {
+  const { data, errors, handleChangeText, handleSubmit, setData, isLoadingOnSubmit } = useForm<User>({
+    initialData: initialSignUpData,
+    validations: signUpValidations,
+    onSubmit: () => {
+      const newId = uuid.v1().toString();
+      const dataWithId: User = { ...data, id: newId };
+      storageManager.addUser(dataWithId);
+      (async function () {
+        await storageManager.saveLoginUserId(newId);
+        replace("Home");
+      })();
+    },
+  });
+
+  const storageManager = AsyncStorageManager.getInstance();
+
+  const [isUsernameUnique, setIsUsernameUnique] = useState(false);
+  const [isUsernameUniqueError, setIsUsernameUniqueError] = useState("");
+
+  const [isEmailUnique, setIsEmailUnique] = useState(false);
+  const [isEmailUniqueError, setIsEmailUniqueError] = useState("");
+
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [confirmPasswordError, setConfirmPasswordError] = useState("");
+
+  const handleChangeGender = (gender: string) => {
+    handleChangeText("gender", gender);
+  };
+
+  const handleConfirmPassword = (pass: string) => {
+    setConfirmPassword(pass);
+    if (pass !== data.password) {
+      setConfirmPasswordError("Password did not match");
+    } else {
+      setConfirmPasswordError("");
+    }
+  };
+
+  const submitVerification = async () => {
+    if (!confirmPassword) {
+      setConfirmPasswordError("Confirm the password");
+      return;
+    }
+
+    if (data.password !== confirmPassword) {
```

Comment:
where are pwd validations??

Reply:
It is present in constants file
due to PR length i did not include that in this PR